Generated build project for legacy solution not working

### Description

The generated NUKE build project for classic .NET framework / legacy project format solutions does not work due to several problems:

1. Missing reference to `Configuration.cs`
2. Wrong target framework version: 4.6.1 instead of 4.7.2
3. Broken version in Nuke.Common reference: `<Reference Include="Nuke.Common, Version=5.0.2MAJOR_MINOR_.0.0, Culture=neutral, processorArchitecture=MSIL">`

### Steps to reproduce

1. Create a fresh .NET project (.NET Framework 4.8)
2. Create NUKE build project with `nuke :setup --boot`
3. Select `.NET Framework/Mono` and `Legacy Format`, otherwise confirm defaults
4. Restore nuget packages in `build\packages.config`
5. Open solution in VisualStudio
6. Compile the generated NUKE build project

### Relevant log output

#### NUKE execution
```
> nuke :setup --boot
NUKE Global Tool version 5.0.2 (Windows,.NETCoreApp,Version=v2.1)
How should the build project be named?
¬  _build
Where should the build project be located?
¬  ./build
What bootstrapping method should be used?
¬  .NET Framework/Mono
What project format should be used?
¬  Legacy Format: supported by all MSBuild/Mono versions
Which NUKE version should be used?
¬  5.0.2 (latest release)
Which solution should be the default?
¬  NukeIt.sln
```

#### Compilation in VisualStudio

```
------ Build started: Project: _build, Configuration: Debug Any CPU ------
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2123,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "Glob, Version=1.0.0.0, Culture=neutral, processorArchitecture=MSIL". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2123,5): warning MSB3106: Assembly strong name "Nuke.Common, Version=5.0.2MAJOR_MINOR_.0.0, Culture=neutral, processorArchitecture=MSIL" is either a path which could not be found or it is a full assembly name which is badly formed. If it is a full assembly name it may contain characters that need to be escaped with backslash(\). Those characters are Equals(=), Comma(,), Quote("), Apostrophe('), Backslash(\).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2123,5): warning MSB3106: Assembly strong name "Nuke.Common, Version=5.0.2MAJOR_MINOR_.0.0, Culture=neutral, processorArchitecture=MSIL" is either a path which could not be found or it is a full assembly name which is badly formed. If it is a full assembly name it may contain characters that need to be escaped with backslash(\). Those characters are Equals(=), Comma(,), Quote("), Apostrophe('), Backslash(\).
C:\Users\stephan.mueller\source\repos\NukeIt\build\Build.cs(3,7,3,11): error CS0246: The type or namespace name 'Nuke' could not be found (are you missing a using directive or an assembly reference?)
C:\Users\stephan.mueller\source\repos\NukeIt\build\Build.cs(4,7,4,11): error CS0246: The type or namespace name 'Nuke' could not be found (are you missing a using directive or an assembly reference?)
C:\Users\stephan.mueller\source\repos\NukeIt\build\Build.cs(5,7,5,11): error CS0246: The type or namespace name 'Nuke' could not be found (are you missing a using directive or an assembly reference?)
C:\Users\stephan.mueller\source\repos\NukeIt\build\Build.cs(6,7,6,11): error CS0246: The type or namespace name 'Nuke' could not be found (are you missing a using directive or an assembly reference?)
C:\Users\stephan.mueller\source\repos\NukeIt\build\Build.cs(7,7,7,11): error CS0246: The type or namespace name 'Nuke' could not be found (are you missing a using directive or an assembly reference?)
C:\Users\stephan.mueller\source\repos\NukeIt\build\Build.cs(8,7,8,11): error CS0246: The type or namespace name 'Nuke' could not be found (are you missing a using directive or an assembly reference?)
C:\Users\stephan.mueller\source\repos\NukeIt\build\Build.cs(9,7,9,11): error CS0246: The type or namespace name 'Nuke' could not be found (are you missing a using directive or an assembly reference?)
C:\Users\stephan.mueller\source\repos\NukeIt\build\Build.cs(10,14,10,18): error CS0246: The type or namespace name 'Nuke' could not be found (are you missing a using directive or an assembly reference?)
C:\Users\stephan.mueller\source\repos\NukeIt\build\Build.cs(11,14,11,18): error CS0246: The type or namespace name 'Nuke' could not be found (are you missing a using directive or an assembly reference?)
C:\Users\stephan.mueller\source\repos\NukeIt\build\Build.cs(12,14,12,18): error CS0246: The type or namespace name 'Nuke' could not be found (are you missing a using directive or an assembly reference?)
C:\Users\stephan.mueller\source\repos\NukeIt\build\Build.cs(15,15,15,24): error CS0246: The type or namespace name 'NukeBuild' could not be found (are you missing a using directive or an assembly reference?)
C:\Users\stephan.mueller\source\repos\NukeIt\build\Build.cs(14,2,14,33): error CS0246: The type or namespace name 'CheckBuildProjectConfigurationsAttribute' could not be found (are you missing a using directive or an assembly reference?)
C:\Users\stephan.mueller\source\repos\NukeIt\build\Build.cs(14,2,14,33): error CS0246: The type or namespace name 'CheckBuildProjectConfigurations' could not be found (are you missing a using directive or an assembly reference?)
C:\Users\stephan.mueller\source\repos\NukeIt\build\Build.cs(30,5,30,11): error CS0246: The type or namespace name 'Target' could not be found (are you missing a using directive or an assembly reference?)
C:\Users\stephan.mueller\source\repos\NukeIt\build\Build.cs(36,5,36,11): error CS0246: The type or namespace name 'Target' could not be found (are you missing a using directive or an assembly reference?)
C:\Users\stephan.mueller\source\repos\NukeIt\build\Build.cs(41,5,41,11): error CS0246: The type or namespace name 'Target' could not be found (are you missing a using directive or an assembly reference?)
C:\Users\stephan.mueller\source\repos\NukeIt\build\Build.cs(26,14,26,27): error CS0246: The type or namespace name 'Configuration' could not be found (are you missing a using directive or an assembly reference?)
C:\Users\stephan.mueller\source\repos\NukeIt\build\Build.cs(28,25,28,33): error CS0246: The type or namespace name 'Solution' could not be found (are you missing a using directive or an assembly reference?)
C:\Users\stephan.mueller\source\repos\NukeIt\build\Build.cs(25,6,25,15): error CS0246: The type or namespace name 'ParameterAttribute' could not be found (are you missing a using directive or an assembly reference?)
C:\Users\stephan.mueller\source\repos\NukeIt\build\Build.cs(25,6,25,15): error CS0246: The type or namespace name 'Parameter' could not be found (are you missing a using directive or an assembly reference?)
C:\Users\stephan.mueller\source\repos\NukeIt\build\Build.cs(28,6,28,14): error CS0246: The type or namespace name 'SolutionAttribute' could not be found (are you missing a using directive or an assembly reference?)
C:\Users\stephan.mueller\source\repos\NukeIt\build\Build.cs(28,6,28,14): error CS0246: The type or namespace name 'Solution' could not be found (are you missing a using directive or an assembly reference?)
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========

```

--- a/source/Nuke.Common/Utilities/TemplateUtility.cs
+++ b/source/Nuke.Common/Utilities/TemplateUtility.cs
@@ -178,7 +178,7 @@ namespace Nuke.Common.Utilities
 
         private static string Replace(this string str, [CanBeNull] IReadOnlyDictionary<string, string> tokens)
         {
-            return tokens?.Aggregate(str, (t, r) => t.Replace($"_{r.Key}_", r.Value));
+            return tokens?.OrderByDescending(x => x.Key.Length).Aggregate(str, (t, r) => t.Replace($"_{r.Key}_", r.Value));
         }
     }
 }

--- a/source/Nuke.GlobalTool/templates/_build.legacy.csproj
+++ b/source/Nuke.GlobalTool/templates/_build.legacy.csproj
@@ -117,6 +117,7 @@
   </ItemGroup>
   <ItemGroup>
     <Compile Include="Build.cs" />
+    <Compile Include="Configuration.cs" />
   </ItemGroup>
   <ItemGroup>
     <None Include="packages.config" />

